17-switch-mass-hypo.md

```diff
@@ -10,5 +10,81 @@ minutes: 10
 
+There are many situations where you want to repurpose a stripping line to look for a new decay that is simliar in topology but dstinct from what was put into the stripping line. The easiest thing to do is, to change some of the hypothesis on particle IDs made in the stripping. This lesson will show you how to do that.
 
+> ## Reinterpreting stripping selections {.callout}
+> Note that with this method you can never find more candidates than the stripping has already found. 
+> However, you can reinterpret parts of the decay to look for new decay modes.
 
+As an example we will switch the decay of the D0 from (K pi) to (pi pi). 
+
+There is an algorithm that allows us to replace parts of the decay descriptor called SubstitutePID:
+
+```python
+# configure an algorithm to substitute the Kaon in the D0-decay by a second pion 
+from Configurables import SubstitutePID
+Subs = SubstitutePID(
+        'MakeD02pipi',
+        Code = "DECTREE('[D*(2010)+ -> (D0 -> K- pi+) pi+]CC')",
+        Substitutions = { # note that SubstitutePID can't handle automatic CC
+        'Charm -> (D0 -> ^K- pi+) Meson' : 'pi-', 
+        'Charm -> (D0 -> ^K+ pi-) Meson' : 'pi+', 
+        }
+)
+```
+
+The algorithm is configured with a name, an initial selection - here simply a decay descriptor and a dictionary specifying the substitutions to be performed. The particles for which the replacement should be done have to be marked with a ^. Note that SubstitutePID does not automatically handle complex conjugation via the CC operator. Therefore you have to specify all substitutions explicitely. 
+
+Next we have to handle the input and output of this algorithm. This is accomplished using [particle selections](https://twiki.cern.ch/twiki/bin/view/LHCb/ParticleSelection). The input to our substitution algorithm will be the candidates produced by the stripping. In order to make them look like a selection we can use the DataOnDemand service:
+```python
+# Stream and stripping line we want to use
+stream = 'AllStreams'
+line = 'D2hhCompleteEventPromptDst2D2RSLine'
+tesLoc = '/Event/{0}/Phys/{1}/Particles'.format(stream, line)
+
+# get the selection(s) created by the stripping
+from PhysSelPython.Wrappers import Selection
+from PhysSelPython.Wrappers import SelectionSequence
+from PhysSelPython.Wrappers import DataOnDemand
+
+strippingSels = [DataOnDemand(Location=tesLoc)] 
+```
+
+The output of the algorithm has to be packaged into a new selection:
+
+```python
+# create a selection using the substitution algorithm
+selSub = Selection("Dst2D0pi_D02pipi_Sel", Algorithm=Subs, RequiredSelections=strippingSels)
+```
+
+Note how the input stripping selection is daisy-chained to the output selection through the RequiredSelections (it has to be a list of selections) argument. The new selection is added into a SelectionSequnce for further use by DaVinci:
+
+```python
+selSeq = SelectionSequence('SelSeq', TopSelection=selSub)
+```
+
```

Comment:
I used underscores in #19 to separate the decays in the `DecayTreeTuple` name. I use camel case for everything else (like `SelSeq` here).